WebPage::mainFrameView() should return a FrameView

charliewolfe opened the pull request:
#### 37a31aff7bcbc1d180255f5b16be3def142ed5b4
<pre>
WebPage::mainFrameView() should return a FrameView
<a href="https://bugs.webkit.org/show_bug.cgi?id=259651">https://bugs.webkit.org/show_bug.cgi?id=259651</a>
rdar://113145893

Reviewed by Alex Christensen.

Make WebPage::mainFrameView() return a FrameView and introduce WebPage::localMainFrameView().

* Source/WebCore/page/Frame.h:
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::snapshotElementRectForScreenshot):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::invalidateContentsForSlowScroll):
(WebKit::WebChromeClient::showPlaybackTargetPicker):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.cpp:
(WebKit::DrawingAreaCoordinatedGraphics::adjustTransientZoom):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp:
(WebKit::LayerTreeHost::LayerTreeHost):
(WebKit::LayerTreeHost::scrollNonCompositedContents):
(WebKit::LayerTreeHost::constrainTransientZoomOrigin const):
(WebKit::LayerTreeHost::layerForTransientZoom const):
* Source/WebKit/WebProcess/WebPage/DrawingArea.cpp:
(WebKit::DrawingArea::mainFrameTiledBacking const):
(WebKit::DrawingArea::prepopulateRectForZoom):
(WebKit::DrawingArea::scaleViewToFitDocumentIfNeeded):
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm:
(WebKit::RemoteLayerTreeDrawingArea::updateGeometry):
(WebKit::RemoteLayerTreeDrawingArea::setViewExposedRect):
(WebKit::RemoteLayerTreeDrawingArea::exposedContentRect const):
(WebKit::RemoteLayerTreeDrawingArea::setExposedContentRect):
(WebKit::RemoteLayerTreeDrawingArea::updateRendering):
* Source/WebKit/WebProcess/WebPage/ViewGestureGeometryCollector.cpp:
(WebKit::ViewGestureGeometryCollector::collectGeometryForSmartMagnificationGesture):
(WebKit::ViewGestureGeometryCollector::collectGeometryForMagnificationGesture):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::setTracksRepaints):
(WebKit::WebPage::isTrackingRepaints const):
(WebKit::WebPage::resetTrackedRepaints):
(WebKit::WebPage::trackedRepaintRects):
(WebKit::WebPage::scalePageInViewCoordinates):
(WebKit::WebPage::setUseFixedLayout):
(WebKit::WebPage::setFixedLayoutSize):
(WebKit::WebPage::fixedLayoutSize const):
(WebKit::WebPage::updateSizeForCSSDefaultViewportUnits):
(WebKit::WebPage::updateSizeForCSSSmallViewportUnits):
(WebKit::WebPage::updateSizeForCSSLargeViewportUnits):
(WebKit::WebPage::setBackgroundColor):
(WebKit::WebPage::updateMainFrameScrollOffsetPinning):
(WebKit::WebPage::mainFrameDidLayout):
(WebKit::WebPage::mainFrameView const):
(WebKit::WebPage::localMainFrameView const):
(WebKit::WebPage::updateIntrinsicContentSizeIfNeeded):
(WebKit::WebPage::scheduleIntrinsicContentSizeUpdate):
(WebKit::WebPage::dispatchDidReachLayoutMilestone):
(WebKit::WebPage::scrollToRect):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::viewportConfigurationChanged):
* Source/WebKit/WebProcess/WebPage/mac/PageBannerMac.mm:
(WebKit::PageBanner::mouseEvent):
* Source/WebKit/WebProcess/WebPage/mac/RemoteLayerTreeDrawingAreaMac.mm:
(WebKit::RemoteLayerTreeDrawingAreaMac::applyTransientZoomToPage):
(WebKit::RemoteLayerTreeDrawingAreaMac::willCommitLayerTree):
* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationDrawingArea.mm:
(WebKit::TiledCoreAnimationDrawingArea::updateRendering):
(WebKit::TiledCoreAnimationDrawingArea::setViewExposedRect):
(WebKit::TiledCoreAnimationDrawingArea::updateGeometry):
(WebKit::TiledCoreAnimationDrawingArea::layerForTransientZoom const):
(WebKit::TiledCoreAnimationDrawingArea::shadowLayerForTransientZoom const):
(WebKit::TiledCoreAnimationDrawingArea::applyTransientZoomToLayers):
(WebKit::TiledCoreAnimationDrawingArea::commitTransientZoom):
(WebKit::TiledCoreAnimationDrawingArea::applyTransientZoomToPage):
* Source/WebKit/WebProcess/WebPage/mac/WKAccessibilityWebPageObjectMac.mm:
(-[WKAccessibilityWebPageObject accessibilityHitTest:]):
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::updatePageScreenProperties):

Canonical link: <a href="https://commits.webkit.org/266456@main">https://commits.webkit.org/266456@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/36e10cd546a3d93425f8728c83ef0cdf87c89ecb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/13853 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/14168 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/14503 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/15591 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/13161 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/13936 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/16676 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/14250 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15835 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/14022 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/14635 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/11740 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/16296 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/11922 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/12500 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/19542 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/12998 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/12664 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/15882 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/13198 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/11076 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/12469 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3365 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/16801 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/13041 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/WebProcess/WebPage/mac/RemoteLayerTreeDrawingAreaMac.mm:
```diff
@@ -71,7 +71,7 @@
 void RemoteLayerTreeDrawingAreaMac::applyTransientZoomToPage(double scale, FloatPoint origin)
 {
-    auto& frameView = *m_webPage.mainFrameView();
+    auto& frameView = *m_webPage.localMainFrameView();
```
Null check, early return.

achristensen07 commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -8799,7 +8805,7 @@ void WebPage::handleContextMenuTranslation(const TranslationContextMenuInfo& inf
 void WebPage::scrollToRect(const WebCore::FloatRect& targetRect, const WebCore::FloatPoint& origin)
 {
-    mainFrameView()->setScrollPosition(IntPoint(targetRect.minXMinYCorner()));
+    localMainFrameView()->setScrollPosition(IntPoint(targetRect.minXMinYCorner()));
```
Null check

achristensen07 commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -6514,8 +6514,14 @@ Frame* WebPage::mainFrame() const
+
+LocalFrameView* WebPage::localMainFrameView() const
 {
     if (auto* frame = dynamicDowncast<LocalFrame>(mainFrame()))
```
This could just return dynamicDowncast<LocalFrameView>(mainFrameView())
That would reduce places in the code where we are trying to do basically the same thing with code that looks a little different.

achristensen07 commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -2461,7 +2461,7 @@ void WebPage::scalePageInViewCoordinates(double scale, IntPoint centerInViewCoor
         return;
 
-    IntPoint scrollPositionAtNewScale = mainFrameView()->rootViewToContents(-centerInViewCoordinates);
+    IntPoint scrollPositionAtNewScale = localMainFrameView()->rootViewToContents(-centerInViewCoordinates);
```
null check, early return.

achristensen07 commented on Source/WebKit/WebProcess/WebPage/ViewGestureGeometryCollector.cpp:
```diff
@@ -251,7 +251,7 @@ void ViewGestureGeometryCollector::computeMinimumAndMaximumViewportScales(double
 void ViewGestureGeometryCollector::collectGeometryForMagnificationGesture()
 {
-    FloatRect visibleContentRect = m_webPage.mainFrameView()->unobscuredContentRectIncludingScrollbars();
+    FloatRect visibleContentRect = m_webPage.localMainFrameView()->unobscuredContentRectIncludingScrollbars();
```
null check, early return.

achristensen07 commented on Source/WebKit/WebProcess/WebPage/ViewGestureGeometryCollector.cpp:
```diff
@@ -114,7 +114,7 @@ void ViewGestureGeometryCollector::collectGeometryForSmartMagnificationGesture(F
 #endif // PLATFORM(IOS_FAMILY)
 
-    IntPoint originInContentsSpace = m_webPage.mainFrameView()->windowToContents(roundedIntPoint(origin));
+    IntPoint originInContentsSpace = m_webPage.localMainFrameView()->windowToContents(roundedIntPoint(origin));
```
null check, early return.

achristensen07 commented on Source/WebKit/WebProcess/WebPage/ViewGestureGeometryCollector.cpp:
```diff
@@ -83,7 +83,7 @@ void ViewGestureGeometryCollector::dispatchDidCollectGeometryForSmartMagnificati
 void ViewGestureGeometryCollector::collectGeometryForSmartMagnificationGesture(FloatPoint origin)
 {
-    FloatRect visibleContentRect = m_webPage.mainFrameView()->unobscuredContentRectIncludingScrollbars();
+    FloatRect visibleContentRect = m_webPage.localMainFrameView()->unobscuredContentRectIncludingScrollbars();
```
null check, early return.

achristensen07 commented on Source/WebKit/WebProcess/WebPage/DrawingArea.cpp:
```diff
@@ -161,14 +161,14 @@ bool DrawingArea::supportsGPUProcessRendering(DrawingAreaType type)
 {
     double currentPageScale = m_webPage.totalScaleFactor();
-    auto* frameView = m_webPage.mainFrameView();
+    auto* frameView = m_webPage.localMainFrameView();
```
null check, early return.

achristensen07 commented on Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp:
```diff
@@ -515,7 +515,7 @@ FloatPoint LayerTreeHost::constrainTransientZoomOrigin(double scale, FloatPoint
 CoordinatedGraphicsLayer* LayerTreeHost::layerForTransientZoom() const
 {
-    RenderLayerBacking* renderViewBacking = m_webPage.mainFrameView()->renderView()->layer()->backing();
+    RenderLayerBacking* renderViewBacking = m_webPage.localMainFrameView()->renderView()->layer()->backing();
```
null check, early return.

achristensen07 commented on Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp:
```diff
@@ -492,7 +492,7 @@ void LayerTreeHost::renderNextFrame(bool forceRepaint)
 FloatPoint LayerTreeHost::constrainTransientZoomOrigin(double scale, FloatPoint origin) const
 {
-    auto& frameView = *m_webPage.mainFrameView();
+    auto& frameView = *m_webPage.localMainFrameView();
```
null check, early return.

achristensen07 commented on Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.cpp:
```diff
@@ -463,7 +463,7 @@ void DrawingAreaCoordinatedGraphics::displayDidRefresh()
 {
     if (!m_transientZoom) {
-        auto& frameView = *m_webPage.mainFrameView();
+        auto& frameView = *m_webPage.localMainFrameView();
```
null check, early return.

achristensen07 commented on Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
```diff
@@ -1518,7 +1518,7 @@ void WebChromeClient::removePlaybackTargetPickerClient(PlaybackTargetClientConte
 void WebChromeClient::showPlaybackTargetPicker(PlaybackTargetClientContextIdentifier contextId, const IntPoint& position, bool isVideo)
 {
-    auto* frameView = m_page.mainFrameView();
+    auto* frameView = m_page.localMainFrameView();
```
null check, early return.

charliewolfe committed:
WebPage::mainFrameView() should return a FrameView
https://bugs.webkit.org/show_bug.cgi?id=259651
rdar://113145893

Reviewed by Alex Christensen.

Make WebPage::mainFrameView() return a FrameView and introduce WebPage::localMainFrameView().

* Source/WebCore/page/Frame.h:
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::snapshotElementRectForScreenshot):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::invalidateContentsForSlowScroll):
(WebKit::WebChromeClient::showPlaybackTargetPicker):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.cpp:
(WebKit::DrawingAreaCoordinatedGraphics::adjustTransientZoom):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp:
(WebKit::LayerTreeHost::LayerTreeHost):
(WebKit::LayerTreeHost::scrollNonCompositedContents):
(WebKit::LayerTreeHost::constrainTransientZoomOrigin const):
(WebKit::LayerTreeHost::layerForTransientZoom const):
* Source/WebKit/WebProcess/WebPage/DrawingArea.cpp:
(WebKit::DrawingArea::mainFrameTiledBacking const):
(WebKit::DrawingArea::prepopulateRectForZoom):
(WebKit::DrawingArea::scaleViewToFitDocumentIfNeeded):
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm:
(WebKit::RemoteLayerTreeDrawingArea::updateGeometry):
(WebKit::RemoteLayerTreeDrawingArea::setViewExposedRect):
(WebKit::RemoteLayerTreeDrawingArea::exposedContentRect const):
(WebKit::RemoteLayerTreeDrawingArea::setExposedContentRect):
(WebKit::RemoteLayerTreeDrawingArea::updateRendering):
* Source/WebKit/WebProcess/WebPage/ViewGestureGeometryCollector.cpp:
(WebKit::ViewGestureGeometryCollector::collectGeometryForSmartMagnificationGesture):
(WebKit::ViewGestureGeometryCollector::collectGeometryForMagnificationGesture):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::setTracksRepaints):
(WebKit::WebPage::isTrackingRepaints const):
(WebKit::WebPage::resetTrackedRepaints):
(WebKit::WebPage::trackedRepaintRects):
(WebKit::WebPage::scalePageInViewCoordinates):
(WebKit::WebPage::setUseFixedLayout):
(WebKit::WebPage::setFixedLayoutSize):
(WebKit::WebPage::fixedLayoutSize const):
(WebKit::WebPage::updateSizeForCSSDefaultViewportUnits):
(WebKit::WebPage::updateSizeForCSSSmallViewportUnits):
(WebKit::WebPage::updateSizeForCSSLargeViewportUnits):
(WebKit::WebPage::setBackgroundColor):
(WebKit::WebPage::updateMainFrameScrollOffsetPinning):
(WebKit::WebPage::mainFrameDidLayout):
(WebKit::WebPage::mainFrameView const):
(WebKit::WebPage::localMainFrameView const):
(WebKit::WebPage::updateIntrinsicContentSizeIfNeeded):
(WebKit::WebPage::scheduleIntrinsicContentSizeUpdate):
(WebKit::WebPage::dispatchDidReachLayoutMilestone):
(WebKit::WebPage::scrollToRect):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::viewportConfigurationChanged):
* Source/WebKit/WebProcess/WebPage/mac/PageBannerMac.mm:
(WebKit::PageBanner::mouseEvent):
* Source/WebKit/WebProcess/WebPage/mac/RemoteLayerTreeDrawingAreaMac.mm:
(WebKit::RemoteLayerTreeDrawingAreaMac::applyTransientZoomToPage):
(WebKit::RemoteLayerTreeDrawingAreaMac::willCommitLayerTree):
* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationDrawingArea.mm:
(WebKit::TiledCoreAnimationDrawingArea::updateRendering):
(WebKit::TiledCoreAnimationDrawingArea::setViewExposedRect):
(WebKit::TiledCoreAnimationDrawingArea::updateGeometry):
(WebKit::TiledCoreAnimationDrawingArea::layerForTransientZoom const):
(WebKit::TiledCoreAnimationDrawingArea::shadowLayerForTransientZoom const):
(WebKit::TiledCoreAnimationDrawingArea::applyTransientZoomToLayers):
(WebKit::TiledCoreAnimationDrawingArea::commitTransientZoom):
(WebKit::TiledCoreAnimationDrawingArea::applyTransientZoomToPage):
* Source/WebKit/WebProcess/WebPage/mac/WKAccessibilityWebPageObjectMac.mm:
(-[WKAccessibilityWebPageObject accessibilityHitTest:]):
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::updatePageScreenProperties):

Canonical link: https://commits.webkit.org/266456@main
